### PR TITLE
Homework.lambda.part1.exercise3 (burachenko)

### PR DESCRIPTION
**Pull request by AlexMem**

### CODE REVIEW

File: src/test/java/lambda/part1/exercise/Exercise3.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise3 {
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortExpressionLambda() {
+        Person[] persons = getPersons();
+
+        // TODO use Arrays.sort + expression-lambda
+        Arrays.sort(persons, (p1, p2) -> p1.getAge() - p2.getAge());
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part1/exercise/Exercise3.java]: It's safer to use Integer.compare because of the edge values

File: src/test/java/lambda/part1/exercise/Exercise3.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        )));
+    }
+
+    @Test
+    @SuppressWarnings("all")
+    void findFirstWithAge30UsingGuavaPredicateLambda() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO use FluentIterable
+        Person person = FluentIterable.from(persons).filter((p) -> p.getAge() == 30).first().get();
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part1/exercise/Exercise3.java]: This way is slower then firstMatch, because you require iterator to go through all of the elements

**Review by vkovrovEPAM — CHANGES_REQUESTED** [review]

File: src/test/java/lambda/part1/exercise/Exercise3.java (hunk 1 above)
Comment (AlexMem) [reply src/test/java/lambda/part1/exercise/Exercise3.java]: I'm afraid person with negative age can be met hardly ever) Thank you for advice, I have changed implementation.

File: src/test/java/lambda/part1/exercise/Exercise3.java (hunk 2 above)
Comment (AlexMem) [reply src/test/java/lambda/part1/exercise/Exercise3.java]: Thank you for advice, search improved.

**Review by vkovrovEPAM — APPROVED** [review]

**Review by AnyaYu — APPROVED** [review]
